feat: Add the support for SWIFT Business Identifier Code (BIC) format (#58)
laminas/laminas-validator: docs/book/validators/bic.md

# Iban Validator

`Laminas\Validator\BusinessIdentifierCode` validates if a given value could be a BIC code. BIC is

> the abbreviation for "Business Identifier Code".

Reviewer: Add ISO number defining the BIC. 
Also, I think, historical reference quote would be appropriate here
```suggestion
`Laminas\Validator\BusinessIdentifierCode` validates if a given value could be
a BIC code as defined by ISO 9362:2014. BIC is the abbreviation for "Business
Identifier Code".

> The original scope of ISO 9362 was limited to the identification of banks;
> consequently, the acronym “BIC” was used as an abbreviation for “bank
> identifier code”. The scope has since been extended, in the first place to
> cover all types of financial institutions, and in the 2009 edition, to cover
> non-financial institutions as well.
>
> --- [ISO 9362:2014 Banking — Banking telecommunication messages — Business
> identifier code (BIC)](https://www.iso.org/standard/60390.htm)
```

Author: I applied your suggestion

# Iban Validator

`Laminas\Validator\BusinessIdentifierCode` validates if a given value could be
a BIC code as defined by ISO 9362:2014. BIC is the abbreviation for "Business
Identifier Code".

> > The original scope of ISO 9362 was limited to the identification of banks;

Reviewer 2: I think we need one empty line before blockquote

Author: added

> # Iban Validator

Reviewer 3: Wrong name in headline!

Author: changed

`Laminas\Validator\BusinessIdentifierCode` validates if a given value could be
a BIC code as defined by ISO 9362:2014. BIC is the abbreviation for "Business
Identifier Code".

> The original scope of ISO 9362 was limited to the identification of banks;
> consequently, the acronym “BIC” was used as an abbreviation for “bank
> identifier code”. The scope has since been extended, in the first place to
> cover all types of financial institutions, and in the 2009 edition, to cover
> non-financial institutions as well.
>
> --- [ISO 9362:2014 Banking — Banking telecommunication messages — Business
> identifier code (BIC)](https://www.iso.org/standard/60390.htm)

> 

Reviewer 3: ```suggestion
`Laminas\Validator\BusinessIdentifierCode` validates if a given value **could be a ["Business Identifier Code" (BIC)](https://www.swift.com/standards/data-standards/bic)** as defined by [ISO 9362](https://wikipedia.org/wiki/ISO_9362).

A BIC is a unique identification code for financial and non-financial institutions.
```

The historical background should not be covered in this documentation, instead a link must be added. The target group of the documentation are the users of the components and therefore the use is in the foreground.

Author: applied

## Supported options

There are no additional supported options for the `BusinessIdentifierCode` validator.

## BIC validation

> BIC codes should be a string which length should be equal to 8 or 11.

Reviewer 3: ```suggestion
BICs should be a string which length should be equal to 8 or 11.
```

Otherwise it means "Business Identifier Code codes". 😉

Author: I have to confess that I have copied the IBAN validator documentation and it uses IBAN number. I will change it

* The 4 first characters can only be letters and it is used to identify a bank or an institution.

* The following 2 characters can only be letters too and it should be a country code assigned within ISO 3166-1 alpha-2. The only exception is the code 'XK' used for the Republic of Kosovo.

* The following 2 characters can be letters or digits. It is used to represent a location (like a city)

> * The last 3 characters are optional and can be letters or digits. It can represent a branche office. The code 'XXX' is often used to represent the ain office when the 11 characters code is used.

Reviewer 3: Please remove empty lines and reduce the line length. In documentation we follow the coding standard:

> * The soft limit on line length MUST be 120 characters.
> * Lines SHOULD NOT be longer than 80 characters; lines longer than that SHOULD be split into multiple subsequent lines of no more than 80 characters each.

https://www.php-fig.org/psr/psr-12/#23-lines

(https://github.com/laminas/laminas-coding-standard/blob/2.0.0rc1/docs/book/v2/intro.md#laminas-coding-standard)

Author: applied coding standards

## Basic usage

```php
$validator = new Laminas\Validator\BusinessIdentifierCode();

if ($validator->isValid($bicCode)) {
    // bic appears to be valid
} else {
    // bic is invalid; print the reasons

> }

Reviewer 3: ```suggestion
if ($validator->isValid('DEUTDEFF')) {
    // Value appears to be valid
}
```

We can use a _real_ example for better demonstration of usage and to allow the example code to be copied.

Author: added a real example
